src/contrib/random/random.cc

```diff
@@ -0,0 +1,148 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file External random functions for tensor.
+ */
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/util.h>
+#include <dmlc/logging.h>
+#include <dmlc/thread_local.h>
+#include <algorithm>
+#include <random>
+#include <ctime>
+
+#define DLPACK_INTEGER_TYPE_SWITCH(type, DType, ...)    \
```

Comment:
Should I put this into dlpack or somewhere else? @tqchen

Reply:
normally it is not very advised to so as a general switch, for now let us keep it local